tools/loader/ImageClassifier.cpp

```diff
@@ -105,9 +124,17 @@ void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
   std::tie(imgHeight, imgWidth, isGray) = getPngInfo(filenames[0].c_str());
   const size_t numChannels = isGray ? 1 : 3;
 
-  // N x C x H x W
-  result->reset(ElemKind::FloatTy,
-                {numImages, numChannels, imgHeight, imgWidth});
+  switch (imageLayout) {
```

Comment:
Could you try the following:
* Remove this switch statement and switch statement in the loop below:
```
for (unsigned z = 0; z < numChannels; z++) {
      for (unsigned y = 0; y < dims[1]; y++) {
        for (unsigned x = 0; x < dims[0]; x++) {
          RH.at({n, numChannels - 1 - z, x, y}) = (imageH.at({x, y, z}));
======         
 switch (imageLayout) {
          case ImageLayout::NCHW:
            RH.at({n, numChannels - 1 - z, x, y}) = (imageH.at({x, y, z}));
            break;
          case ImageLayout::NHWC:
            RH.at({n, x, y, numChannels - 1 - z}) = (imageH.at({x, y, z}));
            break;
          }
=====
```

* Right after loadImagesAndPreprocess call, add another switch statement which does transposition when imageLayout == ImageLayout::NHWC and do nothing for ImageLayout::NCHW.

Would this work?

Reply:
I tried this, I also commented about this to jfix's comment earlier today. the transpose function has a source/dest. So I created another Tensor dataNHWC and tried dataNCHW.transpose(&dataNHWC,NCHW2NHWC) with the NHWC switch after loadimageAndpreprocess call in the main(). But then data is used in onnxmodelloader and in run.inference, where I will have to pass either data or dataNHWC. So that is again 2 more if/else's

Reply:
```
loadImagesAndPreprocess(inputImageFilenames, &data, imageMode);
switch (imageLayout) {
  case ImageLayout::NCHW:
    break;
  case ImageLayout::NHWC: {
    Tensor result;
    data.transpose(&result, NCHWTONHWC);
    data.assign(&result);
    break; 
  }
}
```

Do you need to add any more switches?

Reply:
dataNHWC - sounds like a good idea, and also dataNCHW maybe.